slides/04_quality.qmd

```diff
@@ -275,48 +275,38 @@ handle exceptions and unexpected situations
 
 ```{r, echo = TRUE, eval = FALSE}
 #| code-line-numbers: false
-getMean("a":"z")
+getMean(c("a", "b", "c"))
 ```
 
-[Error in "a":"z" : NA/NaN argument Additionally: Warning messages: In
-getMean("a":"z") : NAs generated by conversion]{.red-small}
+[Error in sum(x) : invalid 'type' (character) of argument]{.red-small}
 :::
 :::
 
 ## Example: CCR#7
 
 ```{r, echo = TRUE, eval = FALSE}
-#| code-line-numbers: "3,5-7,12,14-16,21,27-29"
+#| code-line-numbers: "3,9,15"
 #| style: font-size:0.9em!important;
-# returns the mean of x
+#' returns the mean of x
 getMean <- function(x) {
-    tryCatch({
-        sum(x) / length(x)      
-    }, error = function(e) {
-        stop("Failed to calculate the mean of 'x': ", e$message)
-    })
+    checkmate::assertNumeric(x)
```

Comment:
switched from tryCatch's to checkmate functions because this package is mentioned/recommended in part 3